From c88fe03d01c3b4120fc13e907c3e49ac97887dba Mon Sep 17 00:00:00 2001
From: Christian Lamparter <chunkeey@googlemail.com>
Date: Wed, 17 Aug 2011 13:15:06 +0200
Subject: [PATCH 13/14] carl9170 firmware: add missing out-of-bound check in
 WoWLAN code
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

src/wlan.c: In function ‘wlan_rx_wol_probe_ssid’:
src/wlan.c:830:30: error: variable ‘end’ set but not used

Signed-off-by: Christian Lamparter <chunkeey@googlemail.com>
---
 carlfw/src/wlan.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

--- a/carlfw/src/wlan.c
+++ b/carlfw/src/wlan.c
@@ -845,7 +845,7 @@ static bool wlan_rx_wol_probe_ssid(struc
 		 */
 		data = (u8 *)((struct ieee80211_mgmt *)hdr)->u.probe_req.variable;
 		for (i = 0; i < (unsigned int)(probe_req.ssid_len + 1); i++) {
-			if (scan[i] != data[i])
+			if (data > end || scan[i] != data[i])
 				return false;
 		}
 
